cityofaustin/atd-vz-data
# Invalidate CR3 metadata to cause them to be redownloaded when updated by CRIS

## Associated issues

This PR intends to close https://github.com/cityofaustin/atd-data-tech/issues/14400

## Tag-along back-port

I've back ported a feature from the LDM that allows you to put a CRIS extract in a local directory and have it use that instead of pulling one from the SFTP server. It's super helpful when working on this script, so it felt like a good time to bring it over.

## Testing

* Spin up the local DB and get an export from CRIS. Let me know, and I can wormhole you one. Any old one will work. 
* Replicate production into your local DB
* Put the export in `/atd-vz-data/atd-etl/cris_import/development_extracts/`
* Comment in the `mess_with_incoming_records ...` line around 109 in `cris_import.py`. This will ensure every record is always an update from CRIS. You can try it without that line too.
* From `/atd-vz-data/atd-etl/cris_import` do `docker compose run cris-import`
* While this runs, capture a crash ID that is being updated. This doesn't have to be from the crashes table. For example, it can be the crash_id of a unit that gets an update. Look for something in the STDOUT stream that looks like `Executing update in atd_txdot_crashes for where public.atd_txdot_crashes.crash_id = 19842483` or anything that gives you a crash_id of an imported record from any table.
* Confirm that that record in the `atd_txdot_crashes` table has these fields set this way:
```
cr3_stored_flag = 'N'
cr3_file_metadata = null
cr3_ocr_extraction_date = null
```
* Here's some sql to make that query easier:
```
select crash_id, cr3_stored_flag, cr3_file_metadata, cr3_ocr_extraction_date
          from public.atd_txdot_crashes
          where crash_id = <crash_id>;
```

---

#### Ship list
